ethcore/sync/src/block_sync.rs

```diff
@@ -236,45 +235,39 @@ impl BlockDownloader {
 		let mut valid_response = item_count == 0; //empty response is valid
 		let mut any_known = false;
 		for i in 0..item_count {
-			let info: BlockHeader = r.val_at(i).map_err(|e| {
-				trace!(target: "sync", "Error decoding block header RLP: {:?}", e);
-				BlockDownloaderImportError::Invalid
-			})?;
-			let number = BlockNumber::from(info.number());
+			let info = SyncHeader::from_rlp(r.at(i)?.as_raw().to_vec())?;
```

Comment:
Do you think it's important to keep the logging message? Or will this be logged elsewhere?

Reply:
it's already logged upstream :)